devito/ir/iet/nodes.py

```diff
@@ -407,7 +407,10 @@ def reads(self):
     @cached_property
     def write(self):
         """The Function written by the Expression."""
-        return self.expr.lhs.base.function
+        try:
```

Comment:
what's this for?

Reply:
Oh oups leftover from a different branch. But it's actually needed because currently things like `Eq(BOTTOM, Call(..., writes=x))` fails to catch the writes and leads to issues. Can remove it